rep-2003.rst

```diff
@@ -0,0 +1,76 @@
+REP: 2003
+Title:  Sensor Data and Map QoS Settings
+Author: Steve Macenski <stevenmacenski at gmail.com>
+Status: Draft
+Type: Standards Track
+Content-Type: text/x-rst
+Created: 08-Nov-2019
+ROS-Version: Dashing
+Post-History: 08-Nov-2019
+
+Abstract
+========
+
+This REP covers the standards surrounding Quality of Service (QoS) settings for common needs in ROS2.
+This includes sensor drivers and map representations to standardize interfaces of not only message types outlined by REPs such as REP 118, 138, and 145, but also QoS to ensure ROS2 interoperability.
+
+Specification
+=============
+
+Map Quality of Service
+----------------------
+
+Map providers such as map servers and SLAM implementations are expected to provide all maps over a reliable transient-local topic.
+This is to allow for reliable transmission of map data and provide the map information to any new map clients without delay.
+The depth of the transient-local storage depth is left to the designer, however a single map depth is a reasonable choice for static or globally-updated dynamic map serving applications.
+Map clients are also expected to receive maps over a reliable transient-local topic with variable depth.
+
+This specification does not prescribe any particular map representation but rather applies to all shared map representations, including but not limited to: occupancy grids, feature maps and dense representations.
+
+Sensor Driver Quality of Service
+--------------------------------
+
+Sensor data provided by a sensor driver from a camera, inertial measurement unit, laser scanner, GPS, depth, or range finder are expected to be provided over a `SensorDataQoS` quality of service as provided by the implemented ROS2 version API.
+This is to allow for unreliable transmission of sensor data directly from source to its first processing stage(s).
+These processing stage(s) are also expected to receive a sensor reading using the SensorDataQoS provided QoS profile implemented by the ROS2 version API.
+
+This specification does not enforce QoS settings for sensor data transmitted or received by stage(s) after the sensor driver.
+
+Rationale
+=========
+
+Why Specify QoS Settings
+------------------------
+
+Unlike ROS1, ROS2's QoS settings chosen by an application can affect the ability for data to be transmitted from a publisher to a subscriber.
+There exists several combinations of QoS settings that will not transmit data from one node to another even while composed in a single process.
```

Comment:
I think as specified in https://github.com/ros-infrastructure/rep/pull/212#issuecomment-560525603 , this is too broad a claim and not specific enough.  In particular, I think we should say here that a reliable subscriber won't be able to talk with a best-effort publisher.

Reply:
@clalancette this is just explaining the motivation of the REP, I would think less detail is better so that if things change in the future, we're not trying to play catch up on semantic differences. There are tables in documentation explaining which things play well with each other, I don't think its the role of this document to explain them beyond the fact that such incompatibilities exist.

With that said, if you prefer an example, I can add one. Please let me know your final decision on it wrt this comment.

Reply:
Thinking about it more, I agree with you that the details are too DDS-specific; different RMW backends may have different QoS incompatibilities.

So with that said, I think this sentence should be more like the following:

```
The may exist publisher and subscriber QoS combinations that will not transmit data from one node to another, even while composed in a single process.
```

Reply:
Sounds good to me